<h1>ACTIVITY NO.#1 - PALMER PENGUINS DATA ANALYTICS PROJECT</h1>
<hr>

<h2>Analyst: Arnel N. Caparros</h2>

In [3]:
import numpy as np
import pandas as pd
import warnings
warnings.filterwarnings('ignore')

In [4]:
penguins = pd.read_csv('datasets\\penguins.csv')

In [5]:
penguins

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen            39.1           18.7              181.0   
1       Adelie  Torgersen            39.5           17.4              186.0   
2       Adelie  Torgersen            40.3           18.0              195.0   
3       Adelie  Torgersen             NaN            NaN                NaN   
4       Adelie  Torgersen            36.7           19.3              193.0   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream            55.8           19.8              207.0   
340  Chinstrap      Dream            43.5           18.1              202.0   
341  Chinstrap      Dream            49.6           18.2              193.0   
342  Chinstrap      Dream            50.8           19.0              210.0   
343  Chinstrap      Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN     NaN  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[344 rows x 8 columns]

<h1>Exploratory Data Analysis (EDA)</h1>
<hr>

<h1>1.  What are the species of penguins in the dataset?</h1>

In [7]:
penguins['species']

0         Adelie
1         Adelie
2         Adelie
3         Adelie
4         Adelie
         ...    
339    Chinstrap
340    Chinstrap
341    Chinstrap
342    Chinstrap
343    Chinstrap
Name: species, Length: 344, dtype: object

<h1>Insight No.1</h1>
<p>The species of penguins in the dataset are Adelie, Gentoo, and Chinstrap</p>

<h1>2. How many penguins belong to each species?</h1>

In [8]:
penguins['species'].value_counts()

species
Adelie       152
Gentoo       124
Chinstrap     68
Name: count, dtype: int64

<h1>Insight No.2</h1>
<p>Adelie has 152 penguins, Gentoo has 124 penguins, and Chinstrap has 68 penguiins</p>

<h1>3. How many penguins were observed on each island?</h1>

In [19]:
penguins['island'].value_counts()

island
Biscoe       168
Dream        124
Torgersen     52
Name: count, dtype: int64

<h1>Insight No.3</h1>
<p>In island of Biscoe there was 168 penguins, in Dream ther was 124, and in Torgersen is 52</p>

<h1>4. What is the average bill length (mm) of all penguins?</h1>

In [9]:
penguins['bill_length_mm'].mean()

np.float64(43.9219298245614)

<h1>Insight No.4</h1>
<p>The average bill length (mm) of all penguins is 43.9219298245614</p>

<h1>5. What is the maximum flipper length (mm) recorded?</h1>

In [10]:
penguins['flipper_length_mm'].max()

np.float64(231.0)

<h1>Insight No.5</h1>
<p>The maximum flipper length (mm) recorded is 231.0</p>

<h1>6. What is the average body mass (g) for each penguin species?</h1>

In [11]:
penguins.groupby('species')['body_mass_g'].mean()

species
Adelie       3700.662252
Chinstrap    3733.088235
Gentoo       5076.016260
Name: body_mass_g, dtype: float64

<h1>Insight No.6</h1>
<p>The average body mass (g) for Adelie is 3700.662252, Chinstrap is 3733.088235, and Gentoo is 5076.016260</p>

<h1>7. What is the average flipper length (mm) for each island?</h1>

In [12]:
penguins.groupby('island')['flipper_length_mm'].mean()

island
Biscoe       209.706587
Dream        193.072581
Torgersen    191.196078
Name: flipper_length_mm, dtype: float64

<h1>Insight No.7</h1>
<p>The average flipper length (mm) for Biscoe is 209.706587, Dream is 193.072581, and Torgersen is 191.196078</p>

<h1>8. What is the count of male vs female penguins in each species?</h1>

In [13]:
penguins.groupby(['species', 'sex']).size()

species    sex   
Adelie     female    73
           male      73
Chinstrap  female    34
           male      34
Gentoo     female    58
           male      61
dtype: int64

<h1>Insight No.8</h1>
<p>The count of male vs female penguins in each species</p>
<p>species    sex</p> 
<p>Adelie     female    73</p> 
<p>              male      73</p> 
<p>Chinstrap  female    34</p> 
<p>           male      34</p> 
<p>Gentoo     female    58</p> 
<p>           male      61</p> 

<h1>9. What is the average bill depth (mm) grouped by species and sex?</h1>

In [14]:
penguins.groupby(['species', 'sex'])['bill_depth_mm'].mean()

species    sex   
Adelie     female    17.621918
           male      19.072603
Chinstrap  female    17.588235
           male      19.252941
Gentoo     female    14.237931
           male      15.718033
Name: bill_depth_mm, dtype: float64

<h1>Insight No.9</h1>
<p>The average bill depth (mm) grouped by species and sex</p>
<p>   
Adelie     female    17.621918
           male      19.072603
Chinstrap  female    17.588235
           male      19.252941
Gentoo     female    14.237931
           male      15.718033</p>

<h1>10. Is there a correlation between bill length (mm) and flipper length (mm)?</h1>

In [15]:
penguins[['bill_length_mm', 'flipper_length_mm']].corr()

bill_length_mm  flipper_length_mm
bill_length_mm           1.000000           0.656181
flipper_length_mm        0.656181           1.000000

<h1>Insight No.10</h1>
<p>Yes, there is a correlation between bill length (mm) and flipper length (mm)</p>

<h1>11. Is there a correlation between body mass (g) and flipper length (mm)?</h1>

In [17]:
penguins[['body_mass_g', 'flipper_length_mm']].corr()

body_mass_g  flipper_length_mm
body_mass_g           1.000000           0.871202
flipper_length_mm     0.871202           1.000000

<h1>Insight No.11</h1>
<p>Yes, there is a correlation between body mass (g) and flipper length (mm)</p>

<h1>12. Compare the average body mass (g) of male and female penguins</h1>

In [16]:
penguins.groupby('sex')['body_mass_g'].mean()

sex
female    3862.272727
male      4545.684524
Name: body_mass_g, dtype: float64

<h1>Insight No.12</h1>
<p>The average body mass (g) of male penguin is 4545.684524 and female penguin is 3862.272727</p>